specs/capri/FileShare_Design.md

```diff
@@ -0,0 +1,737 @@
+# Title
+
+**Author(s)**:  [Shruthi MN](https://github.com/Shruthi-1MN), [Pravin Ranjan](https://github.com/PravinRanjan10)
+
+## Summary
+
+A share is a remote, mountable file system. You can mount a share to and access a share from several hosts by several users at a time. we can create a share, list shares, and show information for, update, and delete a share.To create a share, specify one of these supported protocols:
+* NFS
+* CIFS
+
+## Motivation
+
+Multiple users can access the same file share mount point.
+Data that would otherwise be duplicated on each client can be kept in a single location and accessed by client on the network.
+
+### Goals
+
+* Facilitating File Share Service by providing Standard API to manage multiple vendors, simplify File Share API definition  
+* File share across the users based on access capability
+* Scope for now is to support only NFS, SMB file share protocols
+
+### Non-Goals
+
+* Provide support for multiple other file share protocols ex: GlusterFS, HDFS etc.. 
+* Supporting Replication and Data Protection for file share profile
+
+## Design Details
+
+1.  A user can request for action (let’s say: create a share) through
+     Northbound Plugins(NBP). We have already few plugins, but need to
+     modify for file share request.
+2. Now, NBP can communicate to OpenSDS API server.
+3. Based on user request File Share API will be selected.
+4. Internally, OpenSDS File Share API talk to controllers.
+5. Controllers will also maintain the File Share metadata in database.
+6. Controller communicates to Dock, specific driver to create a share.
+
+![FileShare Architecture Diagram](FileShare_Design.png?raw=true "File_Share_Design")
+
+#### Flow diagram:
+![FileShare Architecture Diagram](FileShare_Flow_diagram.png?raw=true "File_Share_Design")
+
+### REST API impact
+
+YES
+Existing Profile API's needs to be changed
+
+### Data model impact
+
+No
```

Comment:
New file share resources, etc. will be stored in the database so there will be data model impact.

Reply:
Yes. I have updated.